src/components/PopupWithForm.js

```diff
@@ -18,46 +22,38 @@ class PopupWithForm extends Popup {
     }, {});
     return userInput;
   }
-  // Comment to the reviewer.
-  // I did not understand the logics of making a method. Could you please explain it?
 
-  // getForm() {
-  // return this._popupForm
-  // }
-  // Also, I did not understand your comment about setting input values using a method.
-  // I get them using getInputValues and then they are passed to the SEL method on line 39
-  // and on 40 it is executed (with the help of callback in index.js)
-  //
-  // //  setInputValues(data) {
-  //   this._inputList.forEach((input) => {
-  //     // Here you insert the `value` by the `name` of the input
-  //     input.value = data[input.name];
-  //   });
-  // }
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._defaultButtonText;
+    }
+  }
 
-  setEventListeners() {
+  // User input values setter
+  setInputValues(data) {
+    this._inputList.forEach((input) => {
+      data[input.name] = input.value;
+    });
+  }
+
+  // Event Listeners setter and submit logics
+  _setEventListeners() {
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
       this._handleFormSubmit(inputValues);
       this._popupForm.reset();
+      this.close();
     });
     super.setEventListeners();
   }
 
+  // Popup Closer
   close() {
-    // Comment to the reviewer - moved the form reset to the SEL method after a submission.
     super.close();
   }
```

Comment:
* You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.

Reply:
done